src/endpoints/authentication/cose_auth.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "ccf/endpoints/authentication/cose_auth.h"
+
+#include "ccf/crypto/cose_verifier.h"
+#include "ccf/crypto/public_key.h"
+#include "ccf/crypto/verifier.h"
+#include "ccf/http_consts.h"
+#include "ccf/pal/locking.h"
+#include "ccf/rpc_context.h"
+#include "ccf/service/tables/members.h"
+#include "ccf/service/tables/users.h"
+#include "ds/lru.h"
+#include "http/http_sig.h"
+
+#include <qcbor/qcbor.h>
+#include <qcbor/qcbor_spiffy_decode.h>
+#include <t_cose/t_cose_sign1_verify.h>
+//#include <t_cose/t_cose_common.h>
+
+namespace ccf
+{
+  namespace cose
+  {
+    static constexpr int64_t COSE_HEADER_PARAM_ALG = 1;
+    static constexpr int64_t COSE_HEADER_PARAM_KID = 4;
+    static constexpr const char* COSE_HEADER_PARAM_MSG_TYPE =
+      "ccf.gov.msg.type";
+    static constexpr const char* COSE_HEADER_PARAM_PROPOSAL_ID =
+      "ccf.gov.msg.proposal_id";
```

Comment:
These look a lot like the names of service tables (especially the `.gov.` part). Is it necessary?

Reply:
This is just cautious namespacing, deliberately very verbose because we expect a ledger auditor to look at those. It's not quite a table name, but hierarchy is close, yes.

Reply:
Verbose is fine but reserving `.gov` for tables will keep it cleaner imo

Reply:
Not sure what you mean by that, these message headers are governance-related/specific, so I think we want to encode that in the name. Messages with a `.gov` header will have their envelope end up in a `.gov` table. Happy to consider other naming options though, but that's my current thinking.